aten/src/ATen/native/quantized/cpu/qlinear_dynamic.cpp

```diff
@@ -70,9 +70,10 @@ class QLinearDynamicInt8 final : public torch::OperatorKernel {
     auto q_params = quant_utils::ChooseQuantizationParams(
         /*min=*/x_min,
         /*max=*/x_max,
-        /*qmin=*/is_signed ? -(1 << (precision - 1)) : 0,
+        /*qmin=*/is_signed ? -(1 << (precision - 1 - 1)) : 0,
```

Comment:
Suggest explicitly overriding quant_min and quant_max as reduce_range is not precise, we are moving to this change across fake-quant and observers.

Reply:
We also need to consider the changes in QNNPACK path as it also use the same `ChooseQuantizationParams` function.